src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -93,5 +97,96 @@ public PagedDebtPositions getPagedDebtPositionsByIngestionFlowFileId(Long ingest
   public void delete(DebtPosition debtPosition) {
     debtPositionDeleteService.delete(debtPosition);
   }
+
+  @Override
+  public Optional<DebtPosition> getDebtPositionByIupdAndOrganizationId(String iupd, Long organizationId) {
+    return debtPositionRepository.findDebtPositionByIupdOrgAndOrganizationId(iupd, organizationId);
+  }
+
+  @Override
+  public void updateDebtPosition(Long debtPositionId, DebtPositionDTO debtPositionDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findEntityGraphByDebtPositionId(debtPositionId);
+    if (debtPosition == null) {
+      throw new EntityNotFoundException("DebtPosition with id %d not found".formatted(debtPositionId));
+    }
+
+    propagateIdsForDebtPosition(debtPosition, debtPositionDTO);
+
+    debtPositionSaveService.saveDebtPositionDTO(debtPositionDTO);
+  }
+
+  private void propagateIdsForDebtPosition(DebtPosition entity, DebtPositionDTO dto) {
+    if (entity == null || dto == null) return;
+
+    if (dto.getDebtPositionId() == null && entity.getDebtPositionId() != null) {
+      dto.setDebtPositionId(entity.getDebtPositionId());
+    }
+
+    Map<Integer, PaymentOption> poByIndex = indexBy(
+      entity.getPaymentOptions(),
+      PaymentOption::getPaymentOptionIndex
+    );
+
+    streamOf(dto.getPaymentOptions())
+      .forEach(poDTO -> {
+        PaymentOption poEntity = poByIndex.get(poDTO.getPaymentOptionIndex());
+        if (poEntity != null) {
+          propagateIdsForPaymentOption(poEntity, poDTO);
+        }
+      });
+  }
+
+  private void propagateIdsForPaymentOption(PaymentOption poEntity, PaymentOptionDTO poDTO) {
+    if (poDTO.getPaymentOptionId() == null && poEntity.getPaymentOptionId() != null) {
+      poDTO.setPaymentOptionId(poEntity.getPaymentOptionId());
+    }
+
+    Map<String, InstallmentNoPII> instByIud = indexBy(
+      poEntity.getInstallments(),
+      InstallmentNoPII::getIud
+    );
+
+    streamOf(poDTO.getInstallments())
+      .forEach(instDTO -> {
+        InstallmentNoPII instEntity = instByIud.get(instDTO.getIud());
+        if (instEntity != null) {
```

Comment:
let's fill the parent entity id
```suggestion
        instDTO.setPaymentOptionId(poDTO.getPaymentOptionId());
        if (instEntity != null) {
```